src/DonorProfiles/resources/js/app/components/text-control/index.js

```diff
@@ -12,7 +12,7 @@ const TextControl = ( { label, value, onChange, icon, type } ) => {
 					<FontAwesomeIcon icon={ icon } />
 				) }
 				<input
-					id={ label && toKebabCase( label ) }
+					id={ label && toKebabCase( `${ label }-${ Math.floor( Math.random() * Math.floor( 1000 ) ) }` ) }
```

Comment:
This logic is duplicated in `select-control`, if these lines need to change for the same reason then they should probably be extracted to a helper method. Otherwise, if each may change independently of the other then the duplicate is the preferred method.

Reply:
Good thought -- I've gone ahead and extracted this into a `toUniqueId` utility. Ref: https://github.com/impress-org/givewp/pull/5486/commits/91a8870d8841d21c0db9a0907a0ac3deb9e70ae4